packages/plugins/plugin-wnfs/src/capabilities/intent-resolver.ts

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import { contributes, Capabilities, createResolver, type PluginsContext } from '@dxos/app-framework';
+import { create } from '@dxos/react-client/echo';
+
+import { WnfsCapabilities } from './capabilities';
+import { upload } from '../helpers';
+import { FileType, WnfsAction } from '../types';
+
+export default (context: PluginsContext) =>
+  contributes(Capabilities.IntentResolver, [
+    // TODO(wittjosiah): Deleting the object doesn't delete the wnfs blob.
+    //  Consider ways to trigger blob deletion based on object deletion and/or adding file system manager.
```

Comment:
Good shout yeah. I should also note that a WNFS never shrinks because it keeps all the history/versions of everything. Also something something about edge blob service keeping all the DAGs. WNFS doesn't have any pruning/snapshots built-in at the moment, but you can kind of replicate that behaviour by creating an entirely new WNFS and adding back the stuff you need (which is doable in this case, as the space just keeps around the access/encryption key).